tinymce-single-react-ui/src/external/wp-blocks/editable/index.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component } from 'wp-elements';
+import tinymce from 'tinymce';
+import { isEqual, last } from 'lodash';
+
+function initialize( node, inline, onSetup ) {
+	if ( ! node ) {
+		return;
+	}
+
+	const config = {
+		target: node.querySelector( '[contenteditable=true]' ),
+		theme: false,
+		inline: true,
+		toolbar: false,
+		skin_url: '//s1.wp.com/wp-includes/js/tinymce/skins/lightgray',
+		entity_encoding: 'raw',
+		setup: onSetup,
+		formats: {
+			strikethrough: { inline: 'del' }
+		}
+	};
+
+	if ( inline ) {
+		config.valid_elements = '#p,br,b,i,strong,em,del,a[href|target]';
+	}
+
+	tinymce.init( config );
+}
+
+export default class EditableComponent extends Component {
+	static defaultProps = {
+		onChange: () => {},
+		splitValue: () => {},
+		onFocusChange: () => {},
+		onType: () => {},
+		initialContent: '',
+		inline: false,
+		single: false,
+	};
+
+	componentDidMount() {
+		initialize( this.node, this.props.inline, this.onSetup );
+		if ( this.props.focusConfig ) {
+			this.focus();
+		}
+	}
+
+	updateContent() {
+		// This could not be called on each content change, it used to change the cursor position
+		let bookmark;
+		if ( this.props.focusConfig ) {
+			bookmark = this.editor.selection.getBookmark( 2, true );
+		}
+		this.editor.setContent( this.props.content );
+		if ( this.props.focusConfig ) {
+			this.editor.selection.moveToBookmark( bookmark );
+		}
+	}
```

Comment:
I was really happy to use your code for the tiny-inside-react component, I think its really well done.
We only had 2 days to work on this prototype so I just hacked the tiny component to do the  minimal changes we needed to get UI state from Tiny. 

My main goal was to show how the UI and UI state can be treated separately from the content state even with the single tiny concept.